### ISSUE TITLE
Individual color spaces not exported

### ISSUE THREAD

**Issue by sgomes** [LeaVerou/color.js#183]
There is currently no mechanism for a consumer to pull in a single color space, as they're not exposed via `/src/index-fn.js`.

We can't expose `spaces/index.js`, since that depends on every color space and expects them all to load. As such, the only feasible option for now is to map them individually in `package.json`, e.g.:

```json
"exports": {
  ".": {
    "import": "./dist/color.js",
    "require": "./dist/color.cjs"
  },
  "./fn": "./src/index-fn.js",
  "./spaces/srgb.js": "./src/spaces/srgb.js",
  "./spaces/p3.js": "./src/spaces/p3.js",
  [...]
}
```

[Node also supports wildcard exports](https://nodejs.org/api/packages.html#subpath-patterns) (which could be a good fit here), but popular bundlers like `webpack` apparently do not, which limits their use.

The alternative would be to work towards dropping side-effects in color space files, so that we could have an index file that would re-export them individually, together with a separate file that would load and register all of them.

I'm happy to contribute the work here, but I'm not sure which of the two options (if either) you think is best!

**Comment by LeaVerou** [comment]
Yeah, right now color spaces cannot be tree-shaken, one would need to import `spaces/index.js`, which as you point out expects them all to load. So this is basically the same issue as #163, just a different facet of it.

I think probably the best way forwards is:
- Color spaces drop side effects
- `spaces/index-fn.js` imports and re-exports all of them
- `index-fn.js` imports and re-exports `spaces/index-fn`
- `spaces/index.js` imports everything from `spaces/index-fn.js` and registers it calling `ColorSpace.register()`

Wondering if there's a better suffix than `fn` for this sort of thing.

**Comment by sgomes** [comment]
> So this is basically the same issue as #163, just a different facet of it.

You're right, this is fundamentally the same issue!

> I think probably the best way forwards is:
> 
> * Color spaces drop side effects
> * `spaces/index-fn.js` imports and re-exports all of them
> * `index-fn.js` imports and re-exports `spaces/index-fn`
> * `spaces/index.js` imports everything from `spaces/index-fn.js` and registers it calling `ColorSpace.register()`

I'm happy to prototype that! I'll post a PR once I have something.

> Wondering if there's a better suffix than `fn` for this sort of thing.

I'm coming up short of alternatives, but I think `fn` is okay, in the same vein as `date-fns` vs more monolithic libraries like moment.

**Comment by LeaVerou** [comment]
Thanks! Since this is a pretty extensive change, you may want to start with a draft PR so you can get feedback early before you make repetitive changes for every color space.

**Comment by sgomes** [comment]
> Thanks! Since this is a pretty extensive change, you may want to start with a draft PR so you can get feedback early before you make repetitive changes for every color space.

Good idea, thank you! 👍

**Comment by LeaVerou** [comment]
This should have been auto closed after #184 was merged, so I'm gonna manual close it.
